lib/blocks.php

```diff
@@ -135,97 +69,30 @@ function get_dynamic_blocks_regex() {
  *
  * @param  array $block A single parsed block object.
  * @return string String of rendered HTML.
  */
 function gutenberg_render_block( $block ) {
-	global $post;
-
-	$block_type    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
-	$is_dynamic    = $block['blockName'] && null !== $block_type && $block_type->is_dynamic();
-	$inner_content = '';
-	$index         = 0;
-
-	foreach ( $block['innerContent'] as $chunk ) {
-		$inner_content .= is_string( $chunk ) ? $chunk : gutenberg_render_block( $block['innerBlocks'][ $index++ ] );
-	}
-
-	if ( $is_dynamic ) {
-		$attributes  = is_array( $block['attrs'] ) ? (array) $block['attrs'] : array();
-		$global_post = $post;
-		$output      = $block_type->render( $attributes, $inner_content );
-		$post        = $global_post;
+	_deprecated_function( __FUNCTION__, '4.9.0', 'render_block()' );
 
-		return $output;
-	}
-
-	return $inner_content;
-}
-
-if ( ! function_exists( 'do_blocks' ) ) {
-	/**
-	 * Parses dynamic blocks out of `post_content` and re-renders them.
-	 *
-	 * @since 0.1.0
-	 * @since 4.4.0 performs full parse on input post content
-	 *
-	 * @param  string $content Post content.
-	 * @return string          Updated post content.
-	 */
-	function do_blocks( $content ) {
-		// If there are blocks in this content, we shouldn't run wpautop() on it later.
-		$priority = has_filter( 'the_content', 'wpautop' );
-		if ( false !== $priority && doing_filter( 'the_content' ) && has_blocks( $content ) ) {
-			remove_filter( 'the_content', 'wpautop', $priority );
-			add_filter( 'the_content', '_restore_wpautop_hook', $priority + 1 );
-		}
-
-		$blocks = gutenberg_parse_blocks( $content );
-		$output = '';
-
-		foreach ( $blocks as $block ) {
-			$output .= gutenberg_render_block( $block );
-		}
-
-		return $output;
-	}
-
-	add_filter( 'the_content', 'do_blocks', 7 ); // BEFORE do_shortcode() and oembed.
-}
-
-if ( ! function_exists( '_restore_wpautop_hook' ) ) {
-	/**
-	 * If do_blocks() needs to remove wpautop() from the `the_content` filter,
-	 * this re-adds it afterwards, for subsequent `the_content` usage.
-	 *
-	 * @access private
-	 *
-	 * @since 4.6.0
-	 *
-	 * @param string $content The post content running through this filter.
-	 * @return string The unmodified content.
-	 */
-	function _restore_wpautop_hook( $content ) {
-		$current_priority = has_filter( 'the_content', '_restore_wpautop_hook' );
-
-		add_filter( 'the_content', 'wpautop', $current_priority - 1 );
-		remove_filter( 'the_content', '_restore_wpautop_hook', $current_priority );
-
-		return $content;
-	}
+	return render_block( $block );
 }
 
 if ( ! function_exists( 'strip_dynamic_blocks' ) ) {
 	/**
 	 * Remove all dynamic blocks from the given content.
 	 *
 	 * @since 3.6.0
+	 * @deprecated 4.9.0
 	 *
 	 * @param string $content Content of the current post.
 	 * @return string
 	 */
 	function strip_dynamic_blocks( $content ) {
+		_deprecated_function( __FUNCTION__, '4.9.0' );
```

Comment:
Is the intent here to remove these in 5.1?

Reply:
> Is the intent here to remove these in 5.1?

That would be the idea, yes.

I can plan to add a note:

https://wordpress.org/gutenberg/handbook/designers-developers/developers/backward-compatibility/deprecations/